fedimint-api/src/core/server.rs

```diff
@@ -69,6 +73,7 @@ pub trait IServerModule: Debug {
         &self,
         dbtx: &mut DatabaseTransaction<'a>,
         consensus_items: Vec<(PeerId, DynModuleConsensusItem)>,
+        module_instance_id: ModuleInstanceId,
```

Comment:
We should not inject module instance id to the modules. In principle they shouldn't need to know, and by not giving it to them we are makinig sure they are oblivious to it and can't do weird stuff with it.

Reply:
It's just the `IServerModule` trait though that no one will ever impl. It doesn't make a huge difference, I'm ok with either.

I think I proposed something along the lines of what got implemented but was thinking about carrying the instance id in the `DynServerModule` struct to avoid having to pass it around.

Reply:
Removed module_instance_id from server.rs where I could. fedimint-server now creates the wrapped DatabaseTransaction